domain/ide/workspace/folder.go

```diff
@@ -553,53 +566,87 @@ func getIssuePerFileFromFlatList(issueList []snyk.Issue) snyk.IssuesByFile {
+	var deltaList []snyk.Issue
+	for filterableIssueType, enabled := range supportedIssueTypes {
+		if !enabled {
+			continue
+		}
+		p := filterableIssueType.ToProduct()
+		deltaIssueByFile, err := f.GetDelta(p)
+		if err == nil {
+			deltaList = append(deltaList, getFlatIssueList(deltaIssueByFile)...)
+		}
+	}
+	return deltaList
+}
+
 func (f *Folder) FilterIssues(issues snyk.IssuesByFile, supportedIssueTypes map[product.FilterableIssueType]bool) snyk.IssuesByFile {
 	logger := f.c.Logger().With().Str("method", "FilterIssues").Logger()
-
 	filteredIssues := snyk.IssuesByFile{}
+
+	if f.c.IsDeltaFindingsEnabled() {
+		issues = getIssuePerFileFromFlatList(f.GetDeltaForAllProducts(supportedIssueTypes))
+	}
+
 	for path, issueSlice := range issues {
 		if !f.Contains(path) {
 			logger.Error().Msg("issue found in cache that does not pertain to folder")
 			continue
 		}
 		for _, issue := range issueSlice {
 			// Logging here will spam the logs
-			if isVisibleSeverity(issue) && supportedIssueTypes[issue.GetFilterableIssueType()] {
+			if isVisibleSeverity(f.c, issue) && supportedIssueTypes[issue.GetFilterableIssueType()] {
 				filteredIssues[path] = append(filteredIssues[path], issue)
 			}
 		}
 	}
 	return filteredIssues
 }
 
-func isVisibleSeverity(issue snyk.Issue) bool {
-	logger := config.CurrentConfig().Logger().With().Str("method", "isVisibleSeverity").Logger()
+func isVisibleSeverity(c *config.Config, issue snyk.Issue) bool {
+	logger := c.Logger().With().Str("method", "isVisibleSeverity").Logger()
 
-	filterSeverity := config.CurrentConfig().FilterSeverity()
+	filterSeverity := c.FilterSeverity()
 	logger.Debug().Interface("filterSeverity", filterSeverity).Msg("Filtering issues by severity")
 
 	switch issue.Severity {
 	case snyk.Critical:
-		return config.CurrentConfig().FilterSeverity().Critical
+		return c.FilterSeverity().Critical
 	case snyk.High:
-		return config.CurrentConfig().FilterSeverity().High
+		return c.FilterSeverity().High
 	case snyk.Medium:
-		return config.CurrentConfig().FilterSeverity().Medium
+		return c.FilterSeverity().Medium
 	case snyk.Low:
-		return config.CurrentConfig().FilterSeverity().Low
+		return c.FilterSeverity().Low
 	}
 	return false
 }
 
-func (f *Folder) publishDiagnostics(product product.Product, issuesByFile snyk.IssuesByFile) {
-	f.sendHovers(issuesByFile)
+func (f *Folder) publishDiagnostics(p product.Product, issuesByFile snyk.IssuesByFile) {
+	f.scanStateAggregator.SummaryEmitter().Emit(f.scanStateAggregator.StateSnapshot())
+	f.sendHovers(p, issuesByFile)
 	f.sendDiagnostics(issuesByFile)
-	f.sendSuccess(product)
+
+	deltaErr := f.hasDeltaError(p)
+	if deltaErr != nil {
+		f.sendScanError(p, deltaErr)
+	} else {
+		f.sendSuccess(p)
+	}
+}
+
+func (f *Folder) hasDeltaError(p product.Product) error {
+	var deltaErr error
+	if f.c.IsDeltaFindingsEnabled() {
+		_, deltaErr = f.scanPersister.GetPersistedIssueList(f.path, p)
```

Comment:
can this be moved to scan aggregator? there we should also have the explicit scan error available?